compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -344,8 +344,9 @@ impl<'a> Parser<'a> {
             match allow_c_variadic {
                 AllowCVariadic::Yes => TyKind::CVarArgs,
                 AllowCVariadic::No => {
-                    // FIXME(Centril): Should we just allow `...` syntactically
+                    // FIXME(c_variadic): Should we just allow `...` syntactically
                     // anywhere in a type and use semantic restrictions instead?
+                    // NOTE: This may regress certain MBE calls if done incorrectly.
```

Comment:
Drive-by update to this FIXME.

`c_variadic` is a better fit here. It can be revisited before its stabilization.

The disclaimer concerns code like

```rs
macro_rules! ck { ($ty:ty) => { compile_error!(""); }; (...) => {}; }
ck!(...);
```

which we don't want to regress. This can easily happen if `can_begin_type` gets updated. If we do want accept `...` given `$ty:ty`, this should probably happen as part of a new edition (cf. fragment specifier `expr_2021`, and `inline_const` in Rust 2024).

Reply:
i feel like `...` is not a real type and we shouldn't treat it like one lol

Reply:
Do u want to add a regression test?

Reply:
Yeah, can do.

Reply:
> i feel like `...` is not a real type and we shouldn't treat it like one in macros lol

Yeah, I feel the same >.<